fvm/evm/emulator/emulator.go

```diff
@@ -522,6 +532,7 @@ func (proc *procedure) run(
 	// if prechecks are passed, the exec result won't be nil
 	if execResult != nil {
 		res.GasConsumed = execResult.UsedGas
+		res.GasRefund = proc.state.GetRefund()
```

Comment:
I would change this into: `res.GasRefund(refund)` that would internally also add to gas consumed

Reply:
That would break out the current functionality. We need to track `GasConsumed` and `GasRefund` separately, and only add them up for `EVM.dryRun`, which is a special case for proper gas estimation.